Lombiq.Tests.UI/Extensions/ExtendedLoggingExtensions.cs

```diff
@@ -104,12 +140,25 @@ public static class ExtendedLoggingExtensions
     private static async Task<TResult> ExecuteSectionAsync<TResult>(
         this UITestContext context, LogSection section, Func<Task<TResult>> functionAsync)
     {
-        // This is somewhat risky. ILogManager is not thread-safe and uses as stack to keep track of sections, so if
-        // multiple sections are started in concurrent threads, the result will be incorrect. This shouldn't be too much
-        // of an issue for now though since tests, while async, are single-threaded.
-        context.Scope.AtataContext.Log.Start(section);
-        var result = await functionAsync();
-        context.Scope.AtataContext.Log.EndSection();
-        return result;
+        for (int i = 0; i < StabilityRetryCount; i++)
+        {
+            var notLast = i < StabilityRetryCount - 1;
+            try
+            {
+                // This is somewhat risky. ILogManager is not thread-safe and uses as stack to keep track of sections, so if
+                // multiple sections are started in concurrent threads, the result will be incorrect. This shouldn't be too much
+                // of an issue for now though since tests, while async, are single-threaded.
+                context.Scope.AtataContext.Log.Start(section);
+                var result = await functionAsync();
+                context.Scope.AtataContext.Log.EndSection();
+                return result;
+            }
+            catch (StaleElementReferenceException) when (notLast)
+            {
+                await Task.Delay(TimeSpan.FromSeconds(1));
+            }
+        }
+
+        throw new InvalidOperationException("Impossible to reach.");
```

Comment:
A custom exception would be better or a constant for the message at least.

Reply:
That would be an unreasonable effort for a theoretically impossible scenario.